pkg/render/common/securitycontext/security_context.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021-2022 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package securitycontext
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/tigera/operator/pkg/ptr"
+)
+
+var (
+	// It is recommended to choose UID and GID that don't collide with existing system users and groups.
+	// Non-system UID and GID range is normally from 1000 to 60000 (Debian derived systems define this
+	// in /etc/login.defs). On a normal Linux host, it is unlikely to have more than 10k non-system users.
+	// 10001 is chosen based on this assumption.
+	RunAsUserID  int64 = 10001
```

Comment:
Preferably, I want this UID and GID consistent across all components. It is not possible to make the Operator only change because some components `chown` folders to a hard-coded UID in `Dockerfile`. When running as a different user, I got permission errors. 10001 is chosen for the reason mentioned in the comments. The UIDs we are currently using (999 or 1001 etc.) may not be desired because they will collide with system or non-system users. Do we want to make them consistent? @caseydavenport or @tmjd WDYT?

Reply:
Made uid and gid configurable in `NewBaseContext`. The UID and GID are taken from Dockerfiles (unchanged) for each component.